##### Key Position #####
key_goal:
  # Key grasping position in map frame [meters]
  x: 0.5715
  y: 0.2623
  z: 0.175 #peg: 0.1665, key: 0.175
  # Key grasping orientation in [radians]
  roll: 3.1416
  pitch: 0.0
  yaw: -0.7854 #Gazebo:0.0, Panda: -0.7854
  # Distance above key_goal_z for pre-grasp [meters]
  z_offset: 0.15

##### Padlock Position #####
padlock_goal:
  # Padlock 'inserted' position in map frame [meters]
  x: 0.5095 #pegboard: 0.5715, padlock: 0.5095
  y: 0.2500 #pegboard: 0.1907, padlock: 0.2500
  z: 0.2134 #pegboard: 0.1665, padlock: 0.2134
  # Padlock insertion orientation [radians]
  roll: 3.1416
  pitch: 0.0
  yaw: -0.7854 #Gazebo:0.0, Panda: -0.7854
  # Distances above padlock_goal_z for pre-insert [meters]
  z_offset_far: 0.15
  z_offset_close: 0.03
  # Intentional misalignment in X for padlock [meters]
  x_misalignment: 0.0
  # Intentional misalignment in Y for padlock [meters]
  y_misalignment: 0.0

##### Home Position #####
home:
  # Home position in joint space [radians]
  j1: 0.0
  j2: -0.785
  j3: 0.0
  j4: -2.356
  j5: 0.0
  j6: 1.57
  j7: 0.784

##### Spiral Insertion #####
spiral:
  # Max. force when moving to insertion plane [Newtons]
  Ft: 1.5
  # Max. wrist force during spiral [Newtons]
  Fd: 4.0
  # Max. insertion force during final insert phase [Newtons]
  Fi: 4.5 #pegboard: 1.5, padlock: 4.5
  # Max. torque in X during spiral [Newton-meters]
  Tx: 0.5
  # Max. Torque in Y during spiral [Newton-meters]
  Ty: 0.5
  # Max. distance for EE to move in Z [meters]
  delta_max: 0.2
  # Distance to move EE in Z each step [meters]
  delta_step: 0.0005
  # Initial distance from spiral origin [meters]
  a: 0.0
  # Distance between spiral turns [meters]
  b: 0.000005
  # Number of points in spiral [-]
  nmax: 1000.0 #Should be double data type
  # Number of spiral rotations [-]
  rot: 10.0 #Should be double data type
  # Number of force-torque samples to take before insertion [-]
  ft_samples: 50
  # Sleep time between force-torque samples [seconds]
  ft_sleep: 0.05
  # Minimum downward (negative) spiral force for detecting padlock/hole [Newtons]
  min_spiral_force: -2.0 # pegboard: -0.6, padlock: -2.0